### ISSUE TITLE
Frontmatter still inserted on template files on save

### ISSUE THREAD

**Issue by aleprovencio** [epwalsh/obsidian.nvim#581]
### 🐛 Describe the bug

Hi!

It seems that `frontmatter` is still being inserted on template files (upon saving), despite this used to be a [fixed issue](https://github.com/epwalsh/obsidian.nvim/issues/141)

My current workaround is to use `disable_frontmatter = true` and manage `frontmatter` for other files on the template file itself, although this is far from optimal

### Config

```lua
opts = {
  workspaces = {
    {
      name = 'zk',
      path = vim.fn.expand '~/.local/share/code/zk-quartz/content',
      overrides = {
        templates = { subdir = 'Templates' },
        --[[ disable_frontmatter = true, ]]
      },
    },
  },
}
```

### Environment

```bash
❯ nvim --version
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1713773202

    arquivo vimrc de sistema: "$VIM/sysinit.vim"
            padrão para $VIM: "/usr/share/nvim"

Run :checkhealth for more info

~ 
❯ nvim --headless -c 'lua require("obsidian").info()' -c q

Obsidian.nvim v3.7.12 (db41b1f20459293436fab510bec58c82a73bd1f7)
Status:
  • buffer directory: nil
  • working directory: /home/aleprovencio
Workspaces:
  ✓ active workspace: Workspace(name='no-vault', path='/home/aleprovencio', root=
'/ho
me/aleprovencio')
  ✗ inactive workspace: Workspace(name='zk', path='/home/aleprovencio/.local/shar
e/co
de/zk-quartz/content', root='/home/aleprovencio/.local/share/code/zk-quartz/conte
nt')
Dependencies:
  ✓ plenary.nvim: 08e301982b9a057110ede7a735dd1b5285eb341f
  ✓ nvim-cmp: cd2cf0c124d3de577fb5449746568ee8e601afc8
Integrations:
  ✓ picker: TelescopePicker()
  ✓ completion: enabled (nvim-cmp) ✗ refs, ✗ tags, ✗ new
    all sources:
      • nvim_lsp
      • path
      • emoji
      • buffer
      • luasnip
Tools:
  ✓ rg: ripgrep 14.1.0
Environment:
  • operating system: Linux
Config:
  • notes_subdir: nil⏎                                                           
```

**Comment by epwalsh** [comment]
@aleprovencio sorry about that, fixed in a9978d0.

**Comment by aleprovencio** [comment]
Thanks @epwalsh !

Could please check whether there's another regression like this on [this issue](https://github.com/epwalsh/obsidian.nvim/issues/218#issuecomment-2113102495) ?
